packages/transformers/src/modules/generic/http_get/http.get.transformer.ts

```
@@ -40,32 +74,32 @@ export class HttpGetTransformer implements ITransformer {
-            } else {
-                const contentType = resp.headers.get('content-type');
-                if (contentType && contentType.includes('application/json')) {
-                    return resp.json();
+            .then(resp => {
+                if (!resp.ok) {
+                    this.telemetryLogger.sendErrorTelemetry(xmsg, `Request failed with code: ${resp.status}`);
+                    throw new Error(`Request failed with code: ${resp.status}`);
                 } else {
-                    return resp.text();
+                    const contentType = resp.headers.get('content-type');
+                    if (contentType && contentType.includes('application/json')) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)